l5kit/l5kit/dataset/agent.py

```diff
@@ -52,17 +52,17 @@ def load_agents_mask(self) -> np.ndarray:
                 "Mask will now be generated for these parameters."
             )
             select_agents(
-                self.dataset.path,
+                self.dataset,
                 agent_prob,
                 history_num_frames,
                 future_num_frames,
                 th_yaw_degree=TH_YAW_DEGREE,
                 th_extent_ratio=TH_EXTENT_RATIO,
                 th_movement=TH_MOVEMENT,
                 th_distance_av=TH_DISTANCE_AV,
-                num_workers=8,
-            )  # TODO maybe set in env var?
-            self.dataset.open()  # ensure root is updated
+                num_workers=8,  # TODO maybe set in env var?
```

Comment:
How about remove it here. Just use the `cpu_count` that `multiprocessing package` gives you for the paralleisation in `select_agents.py`?
`from multiprocessing import cpu_count, Pool`
...
`with Pool(cpu_count()) as pool:`